co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureInstanceManager.java

```diff
@@ -25,20 +25,52 @@
 
 import org.eclipse.mosaic.fed.infrastructure.ambassador.InfrastructureRegistrationMessage;
 import org.eclipse.mosaic.interactions.communication.V2xMessageTransmission;
-import org.eclipse.mosaic.interactions.traffic.VehicleUpdates;
 import org.eclipse.mosaic.lib.enums.AdHocChannel;
-import org.eclipse.mosaic.lib.objects.addressing.AdHocMessageRoutingBuilder;
-import org.eclipse.mosaic.lib.objects.v2x.ExternalV2xContent;
-import org.eclipse.mosaic.lib.objects.v2x.ExternalV2xMessage;
-import org.eclipse.mosaic.lib.objects.v2x.MessageRouting;
-import org.eclipse.mosaic.lib.objects.vehicle.VehicleData;
+import org.eclipse.mosaic.lib.geo.GeoPoint;
 
 /**
  * Session management class for Infrastructure instances communicating with MOSAIC
  * NOTE: TODO See carma.ambassador for reference
  */
 public class InfrastructureInstanceManager {
-    private Map<String, InfrastructureInstance>  managedInstances = new HashMap<>();
+    private Map<String, InfrastructureInstance> managedInstances = new HashMap<>();
     private double currentSimulationTime;
- 
+
+    public void onNewRegistration(InfrastructureRegistrationMessage registration){
+        if (!managedInstances.containsKey(registration.getInfrastructureId())) {
+            try {
```

Comment:
I think this try/catch block is redundant with the one in `newInfrastructureInstance` and can be removed. I believe UnknownHostException is a sub-class of IOException anyway, so the block around `tmp.bind()` would catch it, convert it into a RuntimeException which this block wouldn't catch anyway.

Reply:
This might be needed for the line `InetAddress.getByName(registration.getRxMessageIpAddress())` to see if the IP address is not valid or could not be resolved

Reply:
Ah, okay, that makes more sense then